### PR TITLE
CI: Update test workflow, fix firebase workflow, add Dependabot

### PR DESCRIPTION
**Pull request by EwoutH**
This PR updates the following things:
 - Updates the dependencies in the Gemfile and Gemfile.lock (should fix the firebase workflow)\
 - Adds a [Dependabot](https://dependabot.com/) workflow to automatically open PRs when dependency updates are available (each workday at 8 o'clock).
 - Run the test workflow on all branches to find errors earlier
 - Probably fixes the firebase workflow by updating the dependencies

Consider squashing the commits when merging the PR.

### CODE REVIEW

**Comment by EwoutH** [comment]
The user `dependabot-preview[bot]` needs to be whitelisted in the CLA assistant dashboard for this PR to make sense. See [Can I allow bot user contributions?](https://github.com/cla-assistant/cla-assistant#can-i-allow-bot-user-contributions)

**Comment by ijansch** [comment]
Thanks @EwoutH! I'll add the bot to the whitelist.

**Comment by ijansch** [comment]
Would you be so kind to update the PR by merging in main and resolving the conflicts? Some of the changes conflict with some of the changes that were already underway.

**Comment by EwoutH** [comment]
@ijansch Done!

Feel free to squash this PR when merging.

File: Gemfile [hunk 1]
```diff
@@ -1,8 +1,8 @@
 source 'https://rubygems.org'
 
+gem 'fastlane', '~> 2.166'
 gem 'cocoapods', '~> 1.9'
-gem 'fastlane', '~> 2.1'
 gem 'xcode-install', '~> 2.6'
 
 plugins_path = File.join(File.dirname(__FILE__), 'fastlane', 'Pluginfile')
-eval_gemfile(plugins_path) if File.exist?(plugins_path)
+eval_gemfile('fastlane/Pluginfile') if File.exist?(plugins_path)
```
Comment (7hommay) [inline Gemfile:8]: @EwoutH could you clarify this change?

**Review by 7hommay — COMMENTED** [review]
I've disabled the firebase workflow for now. It shouldn't run from the public repo.

File: Gemfile (hunk 1 above)
Comment (EwoutH) [reply Gemfile:8]: Of course, it fixes this issue with Dependabot: https://github.com/dependabot/dependabot-core/issues/1720#issuecomment-600831687

Comment (7hommay) [reply Gemfile:8]: Thanks!

**Review by 7hommay — APPROVED** [review]

**Comment by 7hommay** [comment]
Thanks @EwoutH !

**Pull request merged**